### DISCUSSION TITLE
StatPanel: allow to group series and show details of members of groups in the tooltip

### DISCUSSION THREAD

**Discussion by w4rgrum** [grafana/grafana#38244] [Dashboards, Panels & Visualizations]
**What would you like to be added**:

I want to have the ability to group multiple series on the same stat box based on a name regexp (in a way similar to the "Rename by regex" transformation feature, but in the panel Options since we do not want to actually drop data, just display them in a grouped way)  and an aggregate action (Max, Min, ...) used to select one value to color/display the grouped box.

![image](https://user-images.githubusercontent.com/54102154/112623589-d0ef9700-8e2c-11eb-8014-070ea65d69fe.png)

Then by having the cursor on a box, the tooltip would show the `X` "first" elements grouped in this box according to options-defined `X` and ordering (Value Increasing, Decreasing, Name ordering, Labels ordering, Threshold color, ...). Ordering could be done on multiple parameters and order, similar to what was possible to do with PolyStat panel (for example threshold decreasing followed by name increasing).

In the example below, I would like to use a regexp gouping on the box names that would use regexp `(...).*` with `$1` as a result value to use (left panel is the raw Stat panel, middle panel is what I expect to get with grouping, right panel is an example of tooltip in the PolyStat panel).

![image](https://user-images.githubusercontent.com/54102154/112623619-da78ff00-8e2c-11eb-92a3-2a704c4d0033.png)

**Why is this needed**:

As you might have guessed this is a feature taken from the PolyStat panel that I feel would make sense to be integrated into the more modern StatPanel, as it allows to have an easy to customize compact display when you have multiple values that could use grouping to be displayed more clearly.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
